wgpu-core/src/device/life.rs

```diff
@@ -545,13 +545,17 @@ impl<A: HalApi> LifetimeTracker<A> {
             &mut trackers.bind_groups,
             |maps| &mut maps.bind_groups,
         );
+        let mut used_buffers = vec![];
+        let mut used_textures = vec![];
```

Comment:
I understand that you're holding these until we're done draining `removed_resource` because you need the strong references in the bind group to go away, to make the `strong_count` check more obviously correct.

But we're already sticking all these buffers and textures in their respective `suspected_resources` tables, which we're guaranteed to scan anyway later in `triage_suspected`. It would be neater to run the `retain` on the buffers' and textures' backlinks from `triage_suspected_buffers` and `triage_suspected_textures`.

Doing it there would have the added benefit that, if there are several bind groups / texture views freed at once, then we can clear out all their `Weak` pointers in a single pass, rather than making a separate pass over the backlinks for each target.